hax/hax/util.py

```diff
@@ -343,9 +344,22 @@ def get_hax_endpoint(self) -> str:
     def get_hax_ip_address(self) -> str:
         return self._service_data().ip_addr
 
+    def get_all_nodes_keys_cached(self) -> Any:
+        return threadkv.get_thread_kv('node_items')
+
+    def reset_all_node_items_cached(self):
+        threadkv.reset_thread_kv('node_items')
+
+    def get_all_nodes_keys(self) -> Any:
+        node_items = threadkv.get_thread_kv('node_items')
```

Comment:
Instead of using `threadkv.get_thread_kv('node_items')`, can we use `get_all_nodes_keys_cached()` ?

And I don't see `get_all_nodes_keys_cached()`  being used anywhere? and similar to get...() do we need a set..() too?

Reply:
Yup, you are right, we can invoke get_all_nodes_keys_cached().
I don't see a need for set_..() for now, but will see.
Thank you.

Reply:
Updated the implementation, kindly review.